src/expr/udf/src/external.rs

```diff
@@ -41,25 +46,40 @@ pub struct ArrowFlightUdfClient {
 impl ArrowFlightUdfClient {
     /// Connect to a UDF service.
     pub async fn connect(addr: &str) -> Result<Self> {
-        let conn = tonic::transport::Endpoint::new(addr.to_string())?
-            .timeout(Duration::from_secs(5))
-            .connect_timeout(Duration::from_secs(5))
-            .connect()
-            .await?;
-        let client = FlightServiceClient::new(conn);
-        Ok(Self {
-            client,
-            addr: addr.into(),
-        })
+        Self::connect_inner(
+            addr,
+            ResolutionStrategy::Eager {
+                timeout: TokioDuration::from_secs(5),
+            },
+        )
+        .await
     }
 
-    /// Connect to a UDF service lazily (i.e. only when the first request is sent).
+    /// Connect to a UDF service lazily (i.e. only fetch IP list from DNS when the first request is sent).
     pub fn connect_lazy(addr: &str) -> Result<Self> {
-        let conn = tonic::transport::Endpoint::new(addr.to_string())?
+        block_on(async { Self::connect_inner(addr, ResolutionStrategy::Lazy).await })
+    }
+
+    async fn connect_inner(addr: &str, resolution_strategy: ResolutionStrategy) -> Result<Self> {
+        let addr = addr.strip_prefix("http://").ok_or_else(|| {
```

Comment:
Any existing udf servers which use `https://`?
Do we enforce this for UDF servers currently? cc @wangrunji0408 . Otherwise during cluster upgrade, users can encounter this error, if they use `https://`.

Reply:
Not sure. But I think both addresses with and without `https://` should be supported.

Reply:
@KeXiangWang can we accept address without `https://` and not return an error here?

Reply:
With the newest update, the url can be either `IP:port`, `http://IP:port` or `https://IP:port`.